volatility3/framework/plugins/linux/mountinfo.py

```diff
@@ -0,0 +1,227 @@
+
+        st_dev = f"{superblock.major}:{superblock.minor}"
+
+        path_root = cls._do_get_path(mnt, task.fs.root)
+        if path_root is None:
+            return None
+
+        mnt_opts: List[str] = []
+        mnt_opts.append(mnt.get_flags_access())
+        mnt_opts.extend(mnt.get_flags_opts())
+
+        # Tagged fields
+        fields: List[str] = []
+        if mnt.is_shared():
+            fields.append(f"shared:{mnt.mnt_group_id}")
+
+        if mnt.is_slave():
+            master = mnt.mnt_master.mnt_group_id
+            fields.append(f"master:{master}")
+            dominating_id = mnt.get_dominating_id(task.fs.root)
+            if dominating_id and dominating_id != master:
+                fields.append(f"propagate_from:{dominating_id}")
+
+        if mnt.is_unbindable():
+            fields.append("unbindable")
+
+        mnt_type = superblock.get_type()
+
+        devname = mnt.get_devname()
+        if not devname:
+            devname = "none"
+
+        sb_opts: List[str] = []
+        sb_opts.append(superblock.get_flags_access())
+        sb_opts.extend(superblock.get_flags_opts())
+
+        return MountInfoData(mnt_id, parent_id, st_dev, mnt_root_path, path_root, mnt_opts, fields, 
+                             mnt_type, devname, sb_opts)
+
+    def _get_mnt_namespace_mountpoints(self, mnt_namespace):
+        mnt_type = self._get_symbol_fullname("mount")
+        if not self.context.symbol_space.has_type(mnt_type):
+            # Old kernels ~ 2.6
+            mnt_type = self._get_symbol_fullname("vfsmount")
+
+        for mount in mnt_namespace.list.to_list(mnt_type, "mnt_list"):
+            yield mount
+            
+    def _get_tasks_mountpoints(self, pids: Iterable[int]):
+        self._vmlinux = self.context.modules[self.config['kernel']]
+               
+        pid_filter = pslist.PsList.create_pid_filter(pids)
+        tasks = pslist.PsList.list_tasks(self.context, self.config['kernel'], filter_func=pid_filter)
+        
+        seen_namespaces = set()
+        for task in tasks:
+            if not (task and task.fs and task.fs.root and task.nsproxy and task.nsproxy.mnt_ns):
+                # This task doesn't have all the information required
+                continue
+
+            mnt_namespace = task.nsproxy.mnt_ns
+            mount_ns_id = mnt_namespace.get_inode()
+            
+            if self._show_mountpoints_per_namespace:
+                if mount_ns_id in seen_namespaces:
+                    continue
+                else:
+                    seen_namespaces.add(mount_ns_id)
+            
+            for mount in self._get_mnt_namespace_mountpoints(mnt_namespace):
+                yield task, mount, mount_ns_id
+
+    def _generator(self):
+        pids = self.config.get('pids')
+
+        for task, mnt, mnt_ns_id in self._get_tasks_mountpoints(pids=pids):
+            mnt_info = self.get_mountinfo(mnt, task)
+            if mnt_info is None:
```

Comment:
I'm not a massive fan of continue unless there's extremely good cause.  It feels here as though the if statement could be inverted and all the following lines indented to achieve the same result.  Again, not a block, just a stylistic thing.  5:)

Reply:
Yeap, sorry. Absolutely, it could be inverted as you said, but I guess it somehow comes to the eternal debate of optimal line limits. In Python, I prefer to return/abort/skip loops/functions asap and avoid nested code contexts. IMO that way helps to keep the code simple, readable and within reasonable line limits. Otherwise, easily you exceed the optimal line length forcing you to massacre your code losing readability. Anyway, probably it's not exactly that case and we can avoid it here. Let me know if you want me change that or otherwise feel free to change that yourself

Reply:
Thanks, I guess I've never read continue statements that often, since I don't write them, so it's a bit jarring to come across them.  Line limits I'm more flexible on unless it goes to three lines or make the thing a mess to read.  I don't think I can easily change the pull request, so best if you do it please.  I'll try to be a little easier on continues where I can...  5;)

Reply:
This is ok if you don't want to fix, it's entirely stylistic.  If it was just forgotten though then now's the time to fix it before I merge...  5;P